python310Packages.pyfuse3: 3.2.3 -> 3.3.0

dotlambda opened the pull request:
## Description of changes
Diff: https://github.com/libfuse/pyfuse3/compare/refs/tags/3.2.3...3.3.0

Changelog: https://github.com/libfuse/pyfuse3/blob/3.3.0/Changes.rst


<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

dotlambda committed:
python310Packages.pyfuse3: 3.2.3 -> 3.3.0

Diff: https://github.com/libfuse/pyfuse3/compare/refs/tags/3.2.3...3.3.0

Changelog: https://github.com/libfuse/pyfuse3/blob/3.3.0/Changes.rst

tjni commented on pkgs/development/python-modules/pyfuse3/default.nix:
```diff
@@ -14,17 +14,17 @@
-  disabled = pythonOlder "3.5";
+  disabled = pythonOlder "3.8";
 
   format = "setuptools";
```
Since v3.2.3, seems like this package uses pyproject.toml. Can we make that switch?

dotlambda replied:
We can, but it's really only a thin wrapper around setuptools in this case.

tjni replied:
Thank you for making the change. You're right that it's just a wrapper, but as I'm doing more work with setuptools, I'm starting to see that it has tons of legacy code paths that are a mess, and I hope moving as much as possible to its PEP 517 code path will be easier to maintain.

tjni commented:
Result of `nixpkgs-review pr 247571` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>borgbackup</li>
    <li>borgbackup.dist</li>
    <li>borgbackup.doc</li>
    <li>borgbackup.man</li>
    <li>borgmatic</li>
    <li>borgmatic.dist</li>
    <li>emborg (python310Packages.emborg)</li>
    <li>emborg.dist (python310Packages.emborg.dist)</li>
    <li>pika-backup</li>
    <li>python310Packages.pyfuse3</li>
    <li>python310Packages.pyfuse3.dist</li>
    <li>python311Packages.emborg</li>
    <li>python311Packages.emborg.dist</li>
    <li>python311Packages.pyfuse3</li>
    <li>python311Packages.pyfuse3.dist</li>
    <li>rmfuse</li>
    <li>rmfuse.dist</li>
    <li>s3ql</li>
    <li>s3ql.dist</li>
    <li>vorta</li>
    <li>vorta.dist</li>
  </ul>
</details>

dotlambda committed:
python310Packages.pyfuse3: use pyproject format

dotlambda committed:
python310Packages.llfuse: 1.4.4 -> 1.5.0

Diff: https://github.com/python-llfuse/python-llfuse/compare/refs/tags/release-1.4.4...1.5.0

Changelog: https://github.com/python-llfuse/python-llfuse/raw/release-1.5.0/Changes.rst

dotlambda commented:
@ofborg build borgbackup
